pkg/core/function.go

```diff
@@ -589,18 +589,34 @@ func (c *client) UpdateFunction(buildpackBuilder Builder, options UpdateFunction
 	return nil
 }
 
+type BuildFunctionOptions struct {
+	BuildOptions
+
+	Image string
+}
+
+func (c *client) BuildFunction(buildpackBuilder Builder, options BuildFunctionOptions, log io.Writer) error {
+	return doBuildLocally(buildpackBuilder, options.Image, options.BuildOptions)
+}
+
 func doBuildLocally(builder Builder, image string, options BuildOptions) error {
+	return doLocally(options, func() error {
+		return builder.Build(options.LocalPath, options.BuildpackImage, options.RunImage, image)
+	})
+}
+
+func doLocally(options BuildOptions, doer func() error) error {
 	if err := writeRiffToml(options); err != nil {
 		return err
 	}
 	defer func() { _ = deleteRiffToml(options) }()
 	if options.BuildpackImage == "" {
-		return fmt.Errorf("unable to build function locally: buildpack image not specified")
+		return fmt.Errorf("unable to do locally: buildpack image not specified")
```

Comment:
`"unable to do locally"` does not seem super clear
what about inlining this function and get back to the previous message: `"unable to build function locally[...]"`